test/services/MeasureService.test.ts

```diff
@@ -345,6 +346,74 @@ describe('MeasureService', () => {
         });
     });
   });
+
+  describe('$data-requirements', () => {
+    // spy on calculation function
+    jest.spyOn(Calculator, 'calculateDataRequirements').mockResolvedValue({
+      results: {
+        resourceType: 'Library',
+        type: {
+          coding: [{ code: 'module-definition', system: 'http://terminology.hl7.org/CodeSystem/library-type' }]
+        },
+        status: 'draft',
+        dataRequirement: []
+      }
+    });
+
+    it('returns 200 and a Library for a simple measure with dependencies and no period params', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Measure/$data-requirements')
+        .send({ resourceType: 'Parameters', parameter: [{ name: 'id', valueString: 'testWithRootLibAndDeps' }] })
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Library');
+          expect(response.body.dataRequirement).toHaveLength(0);
+        });
+    });
+
+    it('returns 200 and a Library for a request with id in the url', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Measure/testWithRootLibAndDeps/$data-requirements')
+        .set('content-type', 'application/fhir+json')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Library');
+          expect(response.body.dataRequirement).toHaveLength(0);
+        });
+    });
+
+    it('returns 200 with passed period parameters', async () => {
```

Comment:
If no period parameters are passed in, the code defaults the periodStart to 2022-01-1 and periodEnd to 2022-12-31. In that case, I think this test might be more beneficial if the passed in periodStart and periodEnd were different (ex. 2018-01-01 and 2018-12-31) if that makes sense!

Reply:
Upon further inspection, I don't think changing this will change anything. Is there any other way to test the periodStart and periodEnd parameters though? Or does fqm-execution mostly change that testing.

Reply:
Y'know. I didn't think there was, but you made me think through it a bit more, and there's totally a way to check the dates using the jest spy. I'll make that change!